src/main/java/graphql/execution/ValuesResolverLegacy.java

```diff
@@ -77,9 +77,6 @@ static Value<?> valueToLiteralLegacy(Object value, GraphQLType type, GraphQLCont
         // Since value is an internally represented value, it must be serialized
         // to an externally represented value before converting into an AST.
         final Object serialized = serializeLegacy(type, value, graphqlContext, locale);
-        if (isNullishLegacy(serialized)) {
-            return null;
-        }
```

Comment:
This NaN check was unnecessary. The NaN check already happened inside the `serialize` method.